internal/batches/ui/task_exec_tui_test.go

```diff
@@ -205,6 +197,70 @@ func TestTaskExecTUI_Integration(t *testing.T) {
 	})
 }
 
+func TestProgressUpdateAfterComplete(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Something emits different escape codes on windows.")
+	}
+
+	buf := &ttyBuf{}
+
+	now := time.Now()
+	clock := func() time.Time { return now.UTC().Truncate(time.Millisecond) }
+
+	out := output.NewOutput(buf, output.OutputOpts{
+		ForceTTY:    true,
+		ForceColor:  true,
+		ForceHeight: 25,
+		ForceWidth:  80,
+		Verbose:     true,
+	})
+
+	tasks := []*executor.Task{
+		{Repository: &graphql.Repository{Name: "github.com/sourcegraph/sourcegraph"}},
+		{Repository: &graphql.Repository{Name: "github.com/sourcegraph/src-cli"}},
+	}
+
+	printer := newTaskExecTUI(out, true, 2)
+	printer.forceNoSpinner = true
+	printer.clock = clock
+
+	// Setup internal state.
+	printer.Start(tasks)
+
+	// Start the tasks.
+	printer.TaskStarted(tasks[0])
+	printer.TaskStarted(tasks[1])
+
+	// Update the tasks into a useful state.
+	printer.TaskCurrentlyExecuting(tasks[0], "echo Hello World > README.md")
+	printer.TaskCurrentlyExecuting(tasks[1], "Downloading archive")
+
+	expectOutput(t, buf, []string{
+		"⠋  Executing... (0/2, 0 errored)  ",
+		"│                                                                               ",
+		"├── github.com/sourcegraph/sourcegraph  echo Hello World > README.md          0s",
+		"└── github.com/sourcegraph/src-cli      Downloading archive                   0s",
+		"",
+	})
+
+	// Now mark the progress as complete.
+	printer.progress.Complete()
+
+	// Now send another update. This would panic before the relevant fix was
+	// merged in #666.
```

Comment:
I just realised we forgot to celebrate #666, the PR of the beast, accordingly (by doing a little ritual in the comments or something) :(

Reply:
From the current changelog:

![image](https://user-images.githubusercontent.com/229984/144088461-9e32638d-33f8-446f-8497-2e36b32bb693.png)

Reply:
Nice, that's the spirit (we summoned)